components/Modal/EditFlightModal.js

```diff
@@ -0,0 +1,104 @@
+import { useState } from "react";
+import { createPortal } from "react-dom";
+
+import {
+  ModalOverlay,
+  ModalContainer,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  CancelButton,
+  SubmitButton,
+  InputLabel,
+  InputField,
+} from "./EditFlightModal.styled";
+
+export default function EditFlightModal({ onClose, selectedFlight, onSave }) {
+  const [flightIata, setFlightIata] = useState(
+    selectedFlight?.flight_iata || ""
+  );
+  const [scheduledDate, setScheduledDate] = useState(
+    selectedFlight?.scheduled_date || ""
+  );
+  const [departure, setDeparture] = useState(selectedFlight?.departure || "");
+  const [arrival, setArrival] = useState(selectedFlight?.arrival || "");
+
+  const handleFlightIataChange = (event) => {
+    setFlightIata(event.target.value);
+  };
+  const handleScheduledDateChange = (event) => {
+    setScheduledDate(event.target.value);
+  };
+
+  const handleDepartureChange = (event) => {
+    setDeparture(event.target.value);
+  };
+
+  const handleArrivalChange = (event) => {
+    setArrival(event.target.value);
+  };
+
+  const handleSaveChanges = () => {
+    event.preventDefault();
+    const editedFlightData = {
+      flight_iata: flightIata,
+      scheduled_date: scheduledDate,
+      departure: departure,
+      arrival: arrival,
+    };
+
+    onSave(editedFlightData);
+    onClose();
+  };
```

Comment:
The event parameter was missing. What is more, I added additional info about the original iata and date of the flight to uniquely identify the entry in local storage. You can now change the iata and date as well.
```suggestion
  const handleSaveChanges = (event) => {
    event.preventDefault();
    const editedFlightData = {
      flight_iata: flightIata,
      scheduled_date: scheduledDate,
      departure: departure,
      arrival: arrival,
    };

    onSave(selectedFlight.flight_iata, selectedFlight.scheduled_date, editedFlightData);
    onClose();
  };
```